### PR TITLE
Restore from severed forked repository.

### PR DESCRIPTION
**Pull request by chrisdoehring**
This is a manual update on a fresh fork for our cellstop integration.
This is meant to re-establish a fork from `gundi-integration-action-runner` template.

Ticket is [GUNDI-3102](https://allenai.atlassian.net/browse/GUNDI-3102)

[GUNDI-3102]: https://allenai.atlassian.net/browse/GUNDI-3102?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by vgarcia13 — DISMISSED** [review]
:shipit:

File: app/actions/configurations.py [hunk 1]
```diff
@@ -1,5 +1,3 @@
-from datetime import datetime
-from typing import List, Optional
 from .core import ActionConfiguration
```
Comment (marianobrc) [inline app/actions/configurations.py]:
```suggestion
from .core import ActionConfiguration,  PullActionConfiguration, AuthActionConfiguration
```

File: app/actions/configurations.py [hunk 2]
```diff
@@ -11,31 +9,5 @@ class AuthenticateConfig(ActionConfiguration):
     refresh_token: str = "string"
 
 
-class PullObservationsHeader(ActionConfiguration):
-    Authorization: str
-
-
 class PullObservationsConfig(ActionConfiguration):
```
Comment (marianobrc) [inline app/actions/configurations.py]:
```suggestion
class PullObservationsConfig(PullActionConfiguration):
```

Comment (marianobrc) [reply app/actions/configurations.py]: we should also make `AuthenticateConfig` to inherit from `AuthActionConfiguration`

File: app/services/activity_logger.py [hunk 3]
```diff
@@ -86,7 +86,7 @@ async def wrapper(*args, **kwargs):
             integration_id = str(integration.id) if integration else None
             action_id = func.__name__.replace("action_", "")
             action_config = kwargs.get("action_config")
-            config_data = action_config.data if action_config else {} or {}
+            config_data = action_config.dict() if action_config else {} or {}
```
Comment (marianobrc) [inline app/services/activity_logger.py]: We shouldn't modify anything within services in the integration. I think you may have found a bug, but then let me fix it in the template repo and we sync the fix later after merging this PR.

File: app/actions/handlers.py [hunk 4]
```diff
@@ -1,28 +1,194 @@
...
+        async for attempt in stamina.retry_context(
+                on=httpx.HTTPError,
+                attempts=3,
+                wait_initial=datetime.timedelta(seconds=10),
+                wait_max=datetime.timedelta(seconds=10),
+        ):
+            with attempt:
+                vehicles = await client.get_vehicles_positions(
+                    integration=integration,
+                    config=client.get_pull_config(integration)
```
Comment (marianobrc) [inline app/actions/handlers.py]: In the latest version of the template now you receive the parsed config in `action_config`, so you don't need to do `client.get_pull_config() `now.
```suggestion
                    config=action_config
```

File: app/actions/handlers.py [hunk 5]
```diff
@@ -1,28 +1,194 @@
...
+
+    return transformed_data
+
+
+async def action_auth(integration, action_config: AuthenticateConfig):
+    logger.info(f"Executing auth action with integration {integration} and action_config {action_config}...")
+    try:
+        token = await client.get_auth_token(
+            integration=integration,
+            config=client.get_auth_config(integration)
```
Comment (marianobrc) [inline app/actions/handlers.py]:
```suggestion
            config=action_config
```
Same as the other comment, now you get what you need in `action_config` :)

File: app/actions/handlers.py [hunk 6]
```diff
@@ -1,28 +1,194 @@
...
+        logger.info(f"Authenticated with success. token: {token}")
+        return {"valid_credentials": token is not None}
+
+
+async def action_fetch_samples(integration, action_config: PullObservationsConfig):
+    logger.info(f"Executing fetch_samples action with integration {integration} and action_config {action_config}...")
+    try:
+        vehicles = await client.get_vehicles_positions(
+            integration=integration,
+            config=client.get_fetch_samples_config(integration)
```
Comment (marianobrc) [inline app/actions/handlers.py]:
```suggestion
            config=action_config
```
Same here

File: app/services/activity_logger.py (hunk 3 above)
Comment (marianobrc) [reply app/services/activity_logger.py]: I opened a PR with the fix:
https://github.com/PADAS/gundi-integration-action-runner/pull/10

File: app/start_scripts/start_app.sh [hunk 7]
```diff
@@ -0,0 +1,11 @@
+#!/bin/bash
```
Comment (marianobrc) [inline app/start_scripts/start_app.sh]: Remove this start script? add it in the template if it's really needed

**Review by marianobrc — APPROVED** [review]
Thank you!

**Pull request merged**